cypress-io/cypress
# feat: experimental single tab run mode for component testing

- Resolves https://github.com/cypress-io/cypress/issues/22353

### User facing changelog


Introduce a new Cypress experiment, `experimentalSingleTabRunMode`, for Component Testing. All specs will execute in a single tab, similar to the Component Testing alpha in Cypress 9.

### Additional details


### Useful Linkd
- [Technical brief](https://github.com/cypress-io/tech-briefs/pull/39/files#diff-f3d6492f15a74d4320ff63d7c537a57327aef8d71cd8e5327c3b553e25f8f098) (Cypress team only)
- [Documentation Update](https://github.com/cypress-io/cypress-documentation/pull/4638/files)
- [Binaries](https://github.com/cypress-io/cypress/commit/293a0f489bbbeae9ecc6d76053fa24eef77fc2de#comments) you can test

In this PR, I added an experimental flag that will reverted back to the Cypress 9 behavior, which will run all the specs for CT in a single tab. It's much faster (see the associated issue for numbers). There are some trade-offs, as noted in the technical brief, which is why this is behind an experimental flag.

This experiment was driven by feedback in https://github.com/cypress-io/cypress/issues/22353, and also from our dashboard users, noting running all CT specs was up to 50% slower. 

Ideally, when we work on the Run All specs feature, we will explore and find out how to best isolate specs running in a single tab. This work will likely make this experiment obsolete, and we'd just deprecate it, in favor of the new run all specs feature. This is a significant project and unlikely to land anytime soon, thus this experiment. 

### Steps to test


- pull this branch down
- go in `packages/launchpad`, which enables the experiment
- `yarn cypress:run:ct --browser chrome --config video=false`. 55s on my machine. :zap: 
- observe it's faster, and it runs all specs in 1 tab. You can add `--headed` to see it running in a single tab.
- remove experimental flag from `cypress.config.ts`
- `yarn cypress:run:ct --browser chrome --config video=false`. 91.9s on my machine. :snail: 

### How has the user experience changed?


Component Tests run much more quickly with this experiment enabled.

### PR Tasks
